chalk-integration/src/program.rs

```diff
@@ -147,6 +147,35 @@ impl tls::DebugContext for Program {
             ParameterKind::Lifetime(n) => write!(fmt, "{:?}", n),
         }
     }
+
+    fn debug_goal(
+        &self,
+        goal: &Goal<ChalkIr>,
+        fmt: &mut fmt::Formatter<'_>,
+    ) -> Result<(), fmt::Error> {
+        let interner = self.interner();
+        match goal.data(interner) {
```

Comment:
Hmm, I realize as I read this -- I think *this* match should live in a `Debug` impl on `GoalData`, probably, which could be in the chalk-ir crate. In other words, this function (and probably the others nearby) should look like

```rust
let interner = self.interner();
write!(fmt, "{:?}", goal.data(interner));
Ok(())
```

Reply:
but I think we should do this in a follow-up PR, I'm going to move this to a separate issue